### PR TITLE
Resolve issue #742 by documenting new hook_civicrm_alterExternUrl

### PR DESCRIPTION
**Pull request by seamuslee001**
ping @kcristiano @eileenmcnaughton @mecachisenros

### CODE REVIEW

**Comment by MikeyMJCO** [comment]
Core PR: https://github.com/civicrm/civicrm-core/pull/15475

**Comment by eileenmcnaughton** [comment]
@MikeyMJCO I think the core pr is merged

**Comment by MikeyMJCO** [comment]
We generally wait for the release before merging into the docs where we can.

File: docs/hooks/hook_civicrm_alterExternUrl.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# hook_civicrm_alterExternUrl
+
+## Summary
+
+This hook allows you to modify extern urls such as click tracking, tracked opens urls
+
```
[the remark this answers is not in the document]
Comment (seamuslee001) [reply docs/hooks/hook_civicrm_alterExternUrl.md]: Fixed

File: docs/hooks/hook_civicrm_alterExternUrl.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- \GuzzleHttp\Psr7\UriInterface `$url` - Object references the passed URL.
+
+- string `$path` - the path to the extern php file without the .php extension e.g.`extern/url`
+
+- string `$query` - URL query parameters as a string
+
+- string `$fragment` - A fragment identifier (anchor) to append to the link)
+
+- bool `$absolute` Whether to force the output to be an absolute link (beginning with a URI-scheme such as 'http:').
```
[the remark this answers is not in the document]
Comment (seamuslee001) [reply docs/hooks/hook_civicrm_alterExternUrl.md]: Fixed

File: docs/hooks/hook_civicrm_alterExternUrl.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+- bool `$isSSL` Whether to redirect to SSL or not format of the URL, NULL to autodetect, TRUE to force to SSL.
```
[the remark this answers is not in the document]
Comment (seamuslee001) [reply docs/hooks/hook_civicrm_alterExternUrl.md]: Thanks @MikeyMJCO resolved now

**Comment by seamuslee001** [comment]
@MikeyMJCO have resolved those 3 things and Core PR went into 5.23 which will be the March release

**Comment by seamuslee001** [comment]
Merging as this has now been released

**Pull request merged**